src/eventstream-proxy/eventstream-proxy.interfaces.ts

```diff
@@ -0,0 +1,57 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { TokenType } from '../tokens/tokens.interfaces';
+
+export interface UriEventData {
+  id: string;
+  value: string;
+}
+
+export interface TransferSingleEventData {
+  from: string;
+  to: string;
+  operator: string;
+  id: string;
+  value: number;
+}
+
+export class TokenPoolEvent {
+  @ApiProperty()
+  pool_id: string;
+
+  @ApiProperty()
+  type: TokenType;
+
+  @ApiProperty()
+  namespace: string;
+
+  @ApiProperty()
+  name: string;
+
+  @ApiProperty()
+  client_id: string;
+}
+
+export class TokenMintEvent {
+  @ApiProperty()
+  pool_id: string;
+
+  @ApiProperty()
+  token_id: string;
```

Comment:
I wonder if this should be `token_index` instead. It's always 0 for fungible pools, and will be a number 1..2^128 for non-fungible pools.